utils/cached_file.hh

```diff
@@ -327,9 +338,11 @@ public:
         size_t count = 0;
         auto disposer = [] (auto* p) noexcept {};
         while (start != end) {
-            if (start->is_linked()) {
+            if (start->unused()) {
                 ++count;
-                _lru.remove(*start);
+                if (start->is_linked()) {
+                    _lru.remove(*start);
+                }
```

Comment:
We should link those pages to LRU instead, so that they don't stay in memory until sstable is deleted.

Reply:
Makes sense, we don't even need to clear the non-lsa buf I think, as we shouldn't penalize the fast path. Just link into LRU, and if no exception happens midway, next read will hit the page in cache and won't have to do any copy (cached_page::get_buf()).